types.cc

```diff
@@ -624,7 +624,13 @@ void write_collection_size(bytes::iterator& out, int size, cql_serialization_for
 }
 
 bytes_view read_collection_value(bytes_view& in, cql_serialization_format sf) {
-    auto size = sf.using_32_bits_for_collections() ? read_simple<int32_t>(in) : read_simple<uint16_t>(in);
+    int32_t size = sf.using_32_bits_for_collections() ? read_simple<int32_t>(in) : read_simple<uint16_t>(in);
```

Comment:
For the v2 protocol (!using_32_bits_for_collections()) I think the sign extension of the uint16_t case is wrong. I think size=65535 meant a null, but your size < 0 check below will not catch it.

Reply:
This text looks like it contains a blatant contradiction. The `[short bytes]` explicitly has the text "if n >= 0", which means there is a possibility that `n < 0` - but this contradicts the definition of `[short]` being an _unsigned_ integer.

So I can only guess (I didn't check) that their actual code was different. Perhaps it was a signed int (but that would limit the value size to 32KB, people would have noticed that), or maybe it was an unsigned int and explicitly compared to 65535 (-1) and not n < 0?